### PR TITLE
feat(application-system): Delete applications

### PR DESCRIPTION
**Pull request by obmagnusson**
# Delete Applications

## What

Allow users to delete their applications in selected states

## Why

Users should be able to delete their drafts

## Screenshots / Gifs
image
image

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by thordurhhh — APPROVED** [review]
service portal lgtm 👍

File: apps/application-system/api/src/app/modules/application/application.controller.ts [hunk 1]
```diff
@@ -1058,4 +1059,35 @@ export class ApplicationController {
...
+    required: true,
+    description: 'The id of the application to delete.',
+    allowEmptyValue: false,
+  })
+  async delete(
+    @Param('id', new ParseUUIDPipe()) id: string,
+    @CurrentUser() user: User,
+  ) {
+    const { nationalId } = user
+    const existingApplication = (await this.applicationAccessService.findOneByIdAndNationalId(
```
Comment (Thorrik58) [inline apps/application-system/api/src/app/modules/application/application.controller.ts:1077]: Should we try to return early here if we know the application doesnt exist for the specific user?

File: apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts [hunk 2]
```diff
@@ -20,6 +26,19 @@ export class ApplicationAccessService {
...
+  }
+
+  async canDeleteApplication(
+    application: Application,
+    nationalId: string,
+  ): Promise<boolean> {
+    const templateId = application.typeId as ApplicationTypes
+    const template = await getApplicationTemplateByTypeId(templateId)
+    const helper = new ApplicationTemplateHelper(application, template)
+    const userRole = template.mapUserToRole(nationalId, application) ?? ''
```
Comment (Thorrik58) [inline apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts:39]: Should we return early if the userRole cant be found e.g. returns ''?

**Review by Thorrik58 — APPROVED** [review]
LGTM, some minor comments that are just a question of readability.

File: libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx [hunk 3]
```diff
@@ -172,7 +200,10 @@ export const ActionCard: React.FC<ActionCardProps> = ({
           <Text variant="small">{date}</Text>
         </Box>
 
-        {!eyebrow && renderTag()}
+        <Inline alignY="center" space={1}>
+          {!eyebrow && renderTag()}
+          {!eyebrow && renderDelete()}
```
Comment (atligudl) [inline libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx:205]: This is confusing, render delete when eyebrow is not defined? But I guess that comes more down to the structure of this component

Sidenote: Rendering React jsx in this way avoides React's optimization services and should generally be avoided.

**Review by atligudl — APPROVED** [review]

**Review by karenbjorg — COMMENTED** [review]
lgtm

**Pull request merged**